### PR TITLE
Replace deprecated functions

### PR DESCRIPTION
**Pull request by Niklp09**
- [x] use mod.conf instead of depends.txt
- [x] replace deprecated functions

### CODE REVIEW

**Comment by S-S-X** [comment]
Maybe also add required engine version to mod.conf, it would be good to have that info because older versions will not be supported anymore.

**Comment by Niklp09** [comment]
5.2?

**Comment by S-S-X** [comment]
> 5.2?

Not sure, I think changed stuff might be all the way from min_minetest_version = 5.0
If you think that 5.2 is closer to correct requirement then just use that otherwise just start from 5.0 because it is good to include it even if not 100% sure what to use. It can be changed later after all if needed, at least there would be version number to blame if there's troubles with API.

**Comment by Niklp09** [comment]
5.0 also works

**Review by S-S-X — APPROVED** [review]

File: mod.conf
Comment (S-S-X) [inline mod.conf]: Patch version number is not documented (and should not be needed for compatibility with semantic versioning), only major and minor `5.0`.
I think it works anyway...

**Comment by S-S-X** [comment]
@Niklp09 invited you to mt-mods organization with write access which means you should be able to merge this yourself if you like.

Generic vague guidelines:
If you're fine with invite and write access just accept, for pull requests it would be good to wait for comments / approval and usually someone will approve things pretty fast around here anyway, at least if PR is somewhat simple and has clear focus on single thing.
Prefer squashing if there's no good reason to move all commits to master.

**Pull request merged**
